helix-core/src/main/java/org/apache/helix/model/PauseSignal.java

```diff
@@ -28,7 +28,11 @@
 public class PauseSignal extends HelixProperty {
 
   public enum PauseSignalProperty {
```

Comment:
I would suggest not mess up with Cluster Status and Action you would like to perform. Better to have two enums to hold them separately. One for user command one for cluster status.

Reply:
We don't want to put cluster status into the PAUSE znode. The PAUSE znode is only created/updated by users. Controller does not write any cluster status to it.
The PAUSE znode will be deleted by controller after cluster fully exits.